### DISCUSSION TITLE
Subscriptions middleware

### DISCUSSION THREAD

**Discussion by mezykr-oanda** [maticzav/graphql-middleware#391] [Q&A]
Hey, first of all thank you for great work on this library! It's working really well for me to intercept resolvers. I'm looking for similar approach to intercept `subscribe` functions for my Subscriptions - is it possible in some way?

**Comment by maticzav** [comment]
Hey! 👋 

Thank you for opening this discussion. I only noticed right now that there's nothing in the documentation regarding subscriptions. As far as I know, they should be covered; I remember we implemented the logic in the codebase. Have you tried adding the middleware using

```ts
const resolvers = {
  Subscribe: middleware,
}
```

If you could create a small sandbox that I can check, we can investigate this a lot more quickly 🙂

**Reply by mezykr-oanda** [reply]
Hey, thank you for your response :)

Here is a link to sandbox: https://codesandbox.io/s/elastic-sun-bxcph?file=/index.js

Please look into terminal, after running `subscription { hello }`, what I'm getting is:

```
🚀 Server ready at http://0.0.0.0:4000/
call Subscription.hello.subscribe with param from middleware undefined
call middleware
call Subscription.hello.resolve with param from middleware HELLO
call middleware
call Subscription.hello.resolve with param from middleware HELLO
call middleware
call Subscription.hello.resolve with param from middleware HELLO
```

What I'm trying to do is to run my middleware for `subscribe` method as well, not only for `resolve`. I wanted to add my extra param to root to be able to see below result in terminal:

```
🚀 Server ready at http://0.0.0.0:4000/
call middleware
call Subscription.hello.subscribe with param from middleware HELLO
call middleware
call Subscription.hello.resolve with param from middleware HELLO
call middleware
call Subscription.hello.resolve with param from middleware HELLO
call middleware
call Subscription.hello.resolve with param from middleware HELLO
```

Thank you!
